token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -50,10 +50,10 @@ public String retrieveTokenKeys(URI tokenKeysEndpointUri, @Nullable String zoneI
 		request.addHeader(HttpHeaders.USER_AGENT, HttpClientUtil.getUserAgent());
 
 		LOGGER.debug("Executing token key retrieval GET request to {} with headers: {} ", tokenKeysEndpointUri,
-				request.getAllHeaders());
-		try (CloseableHttpResponse response = httpClient.execute(request)) {
+				request.headerIterator());
```

Comment:
Not sure if the iterator function is the best replacement here. I will check if getHeaders() is better instead.